Sources/GoodExtensions/Foundation/ArrayExtensions.swift

```diff
@@ -33,140 +34,91 @@ public extension GRActive where Base: Collection {
+    /// Can be used in content factories for UIKit as follows:
+    /// ```swift
+    /// return banners
+    ///     .discardEmpty()?
+    ///     .map { Item.banner($0) }
+    ///     .toArray()
+    ///     .map { Section.banners($0, skeleton: false) } ?? []
+    /// ```
+    /// - Returns: Two-dimensional array with single element that is this array.
+    func toArray() -> [Self] {
+        [self]
     }
-
-    /// This function takes an array of Element elements and chunks it into an array of arrays with a specified size.
-    ///
-    /// - Parameter size: The size of each chunk.
-    /// - Returns: An array of arrays with the chunked elements.
-    func chunked<Element: Strideable>(into size: Int) -> [[Element]] where Base == [Element] {
-        return base.chunked(into: size)
+    
+    /// Appends contents of this array to the end of another array
+    /// - Parameter other: Mutable array where the content should be appended
+    func appendTo(_ other: inout Array) {
+        other.append(contentsOf: self)
     }
-
-    /// Returns an array of unique elements, preserving the original order.
-    ///
-    /// - Returns: An array of unique elements from the original array, preserving the original order.
-    func removedDuplicates<Element: Equatable>() -> [Element] where Base == [Element] {
-        var muttableArray = base
-        let filteredArray = muttableArray.removeDuplicates()
-        return filteredArray
+    
+    /// Replace empty array `[]` with `nil`
+    /// - Returns: `nil` when caller is empty, otherwise unmodified array
+    func discardEmpty() -> Self? {
+        return isEmpty ? nil : self
     }
-
-    /// Prepends a new element to the current array and returns the modified array
-    ///
-    /// - Parameter newElement: The element to be added to the beginning of the array
-    /// - Returns: An array with the new element added to the beginning
-    func prepending<Element: Equatable>(_ newElement: Element) -> [Element] where Base == [Element] {
-        var muttableArray = base
-        muttableArray.insert(newElement, at: 0)
-        return muttableArray
+    
+    /// Replaces array with nil unless the condition is true
+    /// - Parameter condition: condition to check
+    /// - Returns: `nil` when condition is false
+    func `if`(_ condition: Supplier<Bool>) -> Self? {
+        return condition() ? self : nil
+    }
+    
+    /// Prepends another array at the beginning of this array
+    /// - Parameter otherArray: array to prepend
+    /// - Returns: the other array, with contents of this array prepended at the beginning
+    func prepending(_ otherArray: Array<Element>?) -> Self {
+        return (otherArray ?? []) + self
+    }
+    
+    /// Appends another array to the end of this array
+    /// - Parameter otherArray: array to append
+    /// - Returns: this array, with contents of another array appended to the end
+    func appending(_ otherArray: Array<Element>?) -> Self {
```

Comment:
Nebolo by lepsie ak by other array nebol optional?

Reply:
je to ako optional kvoli tym dalsim extensionom `discardEmpty`/`if`, ktore replacuju empty array za `nil`. Na druhej strane ked prependujes nil, tak dostanes to iste pole nad ktorym si to zavolal.

Sample usage:
```swift
let newArray = oldArray
    .map { ... }
    ...
    .prepending(anotherArray.if { userProfile.isSuccess })
    ...
```
(prependujes nieco, pokial plati podmienka, ale vies to napisat takymto kompaktnym sposobom)